neuraxle/hyperparams/space.py

```diff
@@ -74,172 +74,128 @@
 
-    HyperparameterSamples are often the result of calling ``.rvs()`` on an HyperparameterSpace.
+    This class serves as a base for HyperparameterSamples and HyperparameterSpace
     """
     DEFAULT_SEPARATOR = '__'
 
     def __init__(self, *args, separator=None, **kwds):
-        args = self._patch_args(*args)
-
-        if len(args) == 1 and isinstance(args[0], RecursiveDict) and len(kwds) == 0:
-            super().__init__(args[0].items())
-            separator = args[0].separator
-        else:
-            super().__init__(*args, **kwds)
 
         if separator is None:
-            separator = self.DEFAULT_SEPARATOR
-
+            if self._is_only_arg_a_recursive_dict(args, kwds):
+                separator = args[0].separator
+            else:
+                separator = self.DEFAULT_SEPARATOR
         self.separator = separator
 
-    def _patch_args(self, *args):
-        patched_args = []
+        OrderedDict.__init__(self)
         for arg in args:
-            if isinstance(arg, RecursiveDict):
-                patched_arg = RecursiveDict(**{
-                    name: self._patch_arg(value)
-                    for name, value in args[0].items()
-                }, separator=arg.separator)
-            else:
-                patched_arg = arg
-            patched_args.append(patched_arg)
-
-        return patched_args
-
-    def _patch_arg(self, arg):
-        if hasattr(arg, 'dist') and isinstance(arg.dist, rv_generic):
-            if isinstance(arg.dist, rv_discrete):
-                return ScipyDiscreteDistributionWrapper(arg)
-            if isinstance(arg.dist, rv_continuous):
-                return ScipyContinuousDistributionWrapper(arg)
-        else:
-            return arg
-
-    def __getitem__(self, item: str = None):
-        value = self.get(item)
-        if isinstance(value, RecursiveDict) and len(value) == 0:
-            raise ValueError('{} not found in {}'.format(item, self.__class__.__name__))
-        return value
-
-    def get(self, item):
-        item_values = type(self)()
-        for name, values in self.items():
-            if item is None and not self.separator in name:
-                item_values[name] = values
-
-            if self.separator in name:
-                name_split = name.split(self.separator)
-                if str(name_split[0]) == item:
-                    item_values[self.separator.join(name_split[1:])] = values
-
-            if item == name:
-                item_values = values
-        return item_values
-
-    def to_flat(self) -> 'RecursiveDict':
-        """
-        Will create an equivalent flat HyperparameterSamples.
-
-        :return: an HyperparameterSamples like self, flattened.
-        """
-        return self.nested_dict_to_flat(dict_ctor=type(self))
-
-    def to_nested_dict(self) -> 'RecursiveDict':
-        """
-        Will create an equivalent nested dict HyperparameterSamples.
+            self.update(arg)
```

Comment:
There might be a more optimized way to do this with a one-call function to update self after having updated all keys, instead of looping and updating once for each. Optional fix, didn't check really the consequence of what I suggest.